programs/survival/compass.sc

```diff
@@ -0,0 +1,613 @@
+        'spawnWith item <item> <amt> <player> persist' -> ['spawnWithItem', true],
+        'spawnWith effect <effect> <duration> <amplifier>' -> ['spawnWithEffect', 'âny'],
+        'spawnWith effect <effect> <duration> <amplifier> <player>' -> ['spawnWithEffect'],
+        'dontSpawnWith item <item>' -> ['dontSpawnWithItem', 'âny'],
+        'dontSpawnWith item <item> <player>' -> ['dontSpawnWithItem'],
+        'dontSpawnWith effect <effect>' -> ['dontSpawnWithEffect', 'âny'],
+        'dontSpawnWith effect <effect> <player>' -> ['dontSpawnWithEffect'],
+        'timeToTrack <time>' -> ['timeToTrack'],
+        'timeToTrack onUse' -> ['timeToTrack', 'onUse'],
+        'logOnTrack <enabled>' -> ['logOnTrack'],
+        'maxCompasses <amt>' -> ['maxCompasses'],
+        'maxCompasses infinity' -> ['maxCompasses', 'infinity'],
+        'noPortalBreaking <enabled>' -> ['noPortalBreaking'],
+        'maxNetherTravel <dist>' -> ['maxNetherTravel'],
+        'timeToTrack' -> _() -> print(player(), format('l How often the compasses update player positions in seconds: ', 'lb '+(global_timeToTrack/20))),
+        'track' -> _() -> print(player(), format('l Make the compass you\'re holding track a player')),
+        'spawnWith' -> _() -> print(player(), format('l Spawn with a particular item or effect when anyone or a specified player dies')),
+        'dontSpawnWith' -> _() -> print(player(), format('l Stop spawning with a particular item or effect')),
+        'logOnTrack' -> _() -> print(player(), format('l Send a message when someone tracks you: ', 'lb '+if(global_logOnTrack, 'true', 'false'))),
+        'maxCompasses' -> _() -> print(player(), format('l The maximum amount of compasses one can give themselves: ', 'lb '+global_maxCompasses)),
+        'noPortalBreaking' -> _() -> print(player(), format('l Whether to allow breaking portals: ', 'lb '+if(global_noPortalBreaking, 'true', 'false'))),
+        'maxNetherTravel' -> _() -> print(player(), format('l The maximum distance players can travel in the overworld thorugh the nether: ', 'lb '+(global_maxNetherTravel*8))),
+    },
+    'arguments' -> {
+        'player' -> {'type' -> 'string', 'suggester' -> _(args) -> entity_selector('@a'), 'case_sensitive' -> false},
+        'item' -> {'type' -> 'item'},
+        'amt' -> {'type' -> 'int', 'min' -> 1, 'max' -> 64, 'suggest' -> [1, 16, 64]},
+        'time' -> {'type' -> 'float', 'min' -> 0.05, 'suggest' -> [0.5, 1, 10]},
+        'enabled' -> {'type' -> 'bool'},
+        'effect' -> {'type' -> 'effect'},
+        'duration' -> {'type' -> 'int', 'min' -> 0, 'max' -> 1000000, 'suggest' -> [15, 60, 1000000]},
+        'amplifier' -> {'type' -> 'int', 'min' -> 0, 'max' -> 255, 'suggest' -> [0, 1, 2, 3, 255]},
+        'dist' -> {'type' -> 'int', 'min' -> 16, 'max' -> 24000000, 'suggest' -> [8, 128, 7000, 24000000]},
+    }
+};
+
+toKeep = parse_nbt(read_file('toKeep', 'nbt'));
```

Comment:
It might be useful to use nbt_storage() instead 🤔

Reply:
That's probably a good idea